pageserver/src/tenant/layer_map/historic_layer_coverage.rs

```diff
@@ -415,6 +425,46 @@ impl<Value: Clone> BufferedHistoricLayerCoverage<Value> {
         self.buffer.insert(layer_key, None);
     }
 
+    /// Replaces a previous layer with a new layer value if it the given closure returns true.
```

Comment:
```suggestion
    /// Replaces a previous layer with a new layer value if the given closure returns true.
```

Reply:
Well, this probably should be rewritten altogether to include the "unless" cases.

Reply:
[1402120](https://github.com/neondatabase/neon/pull/3513/commits/14021202b7ae5588c8c14760d5da57fce7b0cc4f)